File: 69.py
class Solution:
    def mySqrt(self, x: int) -> int:
        if x == 1: return 1
        minx = 0
        maxx = x
        res = int((minx+maxx)/2)
        while maxx - minx > 1: 
            if res**2 > x:
                maxx = res
                res = int((minx+maxx)/2)
            elif res**2 < x:
                minx = res 
                res = int((minx+maxx)/2)
            else: return res
        return res

print(Solution().mySqrt(8))

File: 69.java
class Solution {
    public int mySqrt(int x) {
        if (x == 0) return 0;
        int res = 1;
        int l = 1, r = x;
        while (l <= r) {
            int mid = l+(r-l)/2;
            if ((double) mid*mid <= x) {
                res = mid;
                l = mid+1;
            } else r = mid-1;
        }
        return res;
    }
}